test/suites/smoke-test-common-solo/test-referenda.ts

```diff
@@ -0,0 +1,72 @@
+
+import { beforeAll, describeSuite, expect } from "@moonwall/cli";
+import type { ApiPromise } from "@polkadot/api";
+
+const AMOUNT_RECENT_PROPOSALS_FOR_CHECKING = 10;
+
+/*
+ * The goal of this smoke test is to ensure that the decision deposit for referenda proposals is set.
+ * If the deposit is not set, the proposal will be in Preparing state until UndecidingTimeout passed (14 days).
+ * Then it will be rejected with event: referenda.TimedOut.
+ */
+describeSuite({
+    id: "SMOK17",
+    title: "Smoke tests for referenda",
+    foundationMethods: "read_only",
+    testCases: ({ it, context, log }) => {
+        let api: ApiPromise;
+
+        beforeAll(async () => {
+            api = context.polkadotJs();
+        });
+
+        it({
+            id: "C01",
+            title: "Referenda deposit does exist for recent proposals",
+            test: async () => {
+                const referendumCount = (await api.query.referenda.referendumCount()).toNumber();
+                const startIndex = Math.max(0, referendumCount - AMOUNT_RECENT_PROPOSALS_FOR_CHECKING);
+
+                for (let proposalIndex = startIndex; proposalIndex < referendumCount; proposalIndex++) {
+                    const infoOpt = await api.query.referenda.referendumInfoFor(proposalIndex);
+                    if (infoOpt.isSome) {
+                        const info = infoOpt.unwrap();
+                        if (info.isOngoing) {
+                            const decisionDeposit = info.asOngoing.decisionDeposit;
+                            const unwrappedDecisionDeposit = decisionDeposit.unwrapOr(null);
+
+                            expect(
+                                unwrappedDecisionDeposit,
+                                `Expecting decision deposit is not null for proposal with index ${proposalIndex}`
+                            ).not.toBeNull();
+                        } else if (info.isApproved) {
+                            const atBlock = info.asApproved[0].toNumber();
+                            await checkDecisionDepositAtBlock(api, proposalIndex, atBlock - 1);
+                        } else if (info.isRejected) {
+                            const atBlock = info.asRejected[0].toNumber();
+                            await checkDecisionDepositAtBlock(api, proposalIndex, atBlock - 1);
+                        }
+                    }
+                }
+            },
+        });
+    },
+});
+
+export async function checkDecisionDepositAtBlock(
+    api: ApiPromise,
+    referendaIndex: number,
+    atBlock: number
+): Promise<void> {
+    const blockHash = await api.rpc.chain.getBlockHash(atBlock);
+    const apiAt = await api.at(blockHash);
+
+    const optInfo = await apiAt.query.referenda.referendumInfoFor(referendaIndex);
+    const info = optInfo.unwrap();
+
+    expect(!!info).toEqual(true);
+    expect(info.isOngoing).toEqual(true);
+    expect(info.asOngoing.decisionDeposit.isSome).toEqual(true);
```

Comment:
Should we also check if decision deposit is equal to that track's decision deposit?

Reply:
Yes, I will add it.

Reply:
Added deposit check: https://github.com/moondance-labs/tanssi/pull/1183/commits/f1bb99748948f8bdd926177e4f29cc967d5cfa98